sdk/core/core-client-rest/src/common.ts

```diff
@@ -48,8 +53,129 @@ export type RequestParameters = {
    * Path parameters for custom the base url
    */
   pathParameters?: Record<string, any>;
+
+  /**
+   * The number of milliseconds a request can take before automatically being terminated.
+   */
+  timeout?: number;
+
+  /**
+   * Callback which fires upon upload progress.
+   */
+  onUploadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * Callback which fires upon download progress.
+   */
+  onDownloadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * The signal which can be used to abort requests.
+   */
+  abortSignal?: AbortSignalLike;
+
+  /**
+   * Options used when tracing is enabled.
+   */
+  tracingOptions?: OperationTracingOptions;
+
+  /**
+   * A function to be called each time a response is received from the server
+   * while performing the requested operation.
+   * May be called multiple times.
+   */
+  onResponse?: RawResponseCallback;
 };
 
+/**
+ * A function to be called each time a response is received from the server
+ * while performing the requested operation.
+ * May be called multiple times.
+ */
+export type RawResponseCallback = (rawResponse: FullOperationResponse, error?: unknown) => void;
+
+/**
+ * Wrapper object for http request and response. Deserialized object is stored in
+ * the `parsedBody` property when the response body is received in JSON or XML.
+ */
+export interface FullOperationResponse extends PipelineResponse {
+  /**
+   * The parsed HTTP response headers.
+   */
+  rawHeaders?: RawHttpHeaders;
+
+  /**
+   * The response body as parsed JSON or XML.
+   */
+  parsedBody?: RequestBodyType;
+
+  /**
+   * The request that generated the response.
+   */
+  request: PipelineRequest;
+}
+
+/**
+ * The base options type for all operations.
+ */
+export interface OperationOptions {
+  /**
+   * The signal which can be used to abort requests.
+   */
+  abortSignal?: AbortSignalLike;
+  /**
+   * Options used when creating and sending HTTP requests for this operation.
+   */
+  requestOptions?: OperationRequestOptions;
+  /**
+   * Options used when tracing is enabled.
+   */
+  tracingOptions?: OperationTracingOptions;
+
+  /**
+   * A function to be called each time a response is received from the server
+   * while performing the requested operation.
+   * May be called multiple times.
+   */
+  onResponse?: RawResponseCallback;
+}
+
+/**
+ * Options used when creating and sending HTTP requests for this operation.
+ */
+export interface OperationRequestOptions {
+  /**
+   * User defined custom request headers that
+   * will be applied before the request is sent.
+   */
+  headers?: RawHttpHeadersInput;
```

Comment:
Just realize the original type of customHeader is `{ [key: string]: string }`, which will not take number or boolean type value, so I change the type of headers into RawHttpHeadersInput which is `Record<string, string | number | boolean>`. it should be consistent with what's in the `RequestParameters`.